pkg/detectors/salesmate/salesmate.go

```diff
@@ -0,0 +1,85 @@
+package salesmate
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/common"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/detectors"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/pb/detectorspb"
+)
+
+type Scanner struct{}
+
+// Ensure the Scanner satisfies the interface at compile time
+var _ detectors.Detector = (*Scanner)(nil)
+
+var (
+	client = common.SaneHttpClient()
+
+	//Make sure that your group is surrounded in boundry characters such as below to reduce false positives
+	keyPat = regexp.MustCompile(detectors.PrefixRegex([]string{"salesmate"}) + `\b([0-9Aa-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})\b`)
+	domainPat = regexp.MustCompile(detectors.PrefixRegex([]string{"salesmateDomain"}) + `\b([a-z0-9A-Z._]{4,22})\b`)
+)
+
+// Keywords are used for efficiently pre-filtering chunks.
+// Use identifiers in the secret preferably, or the provider name.
+func (s Scanner) Keywords() []string {
+	return []string{"salesmate"}
+}
+
+// FromData will find and optionally verify Salesmate secrets in a given set of bytes.
+func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (results []detectors.Result, err error) {
+	dataStr := string(data)
+
+	matches := keyPat.FindAllStringSubmatch(dataStr, -1)
+	idmatches := domainPat.FindAllStringSubmatch(dataStr, -1)
+
+	for _, match := range matches {
+		if len(match) != 2 {
+			continue
+		}
+		resMatch := strings.TrimSpace(match[1])
+		for _, idmatch := range idmatches {
+			if len(idmatch) != 2 {
+				continue
+			}
+			resIdMatch := strings.TrimSpace(idmatch[1])
+			s1 := detectors.Result{
+				DetectorType: detectorspb.DetectorType_Salesmate,
+				Raw:          []byte(resMatch),
+			}
+	
+			if verify {
+				url := fmt.Sprintf("https://%s.salesmate.io/apis/v3/companies/1?trackingRecentSearch=true",resIdMatch)
+				req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
+				if err != nil {
+					continue
+				}
+				
+				req.Header.Add("Content-Type", "application/json")
+				req.Header.Add("sessionToken", fmt.Sprintf("%s", resMatch))
```

Comment:
nit: Prefer removal of fmt.Sprintf here: `req.Header.Add("sessionToken", resMatch)`